app/src/FreeDeck.cpp

```diff
@@ -34,324 +30,214 @@ unsigned long timeOutStartTime;
-			}
-			executeButtonConfig(buttonIndex, state, 0);
-		}
-	}
-	buttonIsUp[buttonIndex] = state;
+  setMuxAddress(buttonIndex, TYPE_BUTTON);
+  uint8_t state = digitalRead(BUTTON_PIN);
+  buttons[buttonIndex].update(state);
+  return;
 }
 
 void initAllDisplays() {
-	for (uint8_t buttonIndex = 0; buttonIndex < BD_COUNT; buttonIndex++) {
-		buttonIsUp[buttonIndex] = 1;
-		downTime[buttonIndex] = 0;
-		longPressed[buttonIndex] = 0;
-		setMuxAddress(buttonIndex, TYPE_DISPLAY);
-		delay(1);
-		oledInit(0x3c, 0, 0);
-		oledFill(255);
-	}
+  for (uint8_t buttonIndex = 0; buttonIndex < BD_COUNT; buttonIndex++) {
+    buttons[buttonIndex].index = buttonIndex;
+    setMuxAddress(buttonIndex, TYPE_DISPLAY);
+    delay(1);
+    oledInit(0x3c, 0, 0);
+    oledFill(255);
+  }
 }
 
 void loadConfigFile() {
-	configFile = SD.open(CONFIG_NAME, FILE_READ);
-	configFile.seek(2);
-	configFile.read(&fileImageDataOffset, 2);
-	pageCount = (fileImageDataOffset - 1) / BD_COUNT;
-	fileImageDataOffset = fileImageDataOffset * 16;
+  configFile = SD.open(CONFIG_NAME, FILE_READ);
+  configFile.seek(2);
+  configFile.read(&fileImageDataOffset, 2);
+  pageCount = (fileImageDataOffset - 1) / BD_COUNT;
+  fileImageDataOffset = fileImageDataOffset * 16;
 }
 
 void initSdCard() {
-	int i = 0;
-	//, SD_SCK_MHZ(50)
-	while (!SD.begin(SD_CS_PIN, SD_SCK_MHZ(16)) && i <= 100) {
-		i++;
-	}
-	if (i == 100) {
-		while (1)
-			;
-	}
-}
-
-void dumpConfigFileOverSerial() {
-	configFile.seekSet(0);
-	if (configFile.available()) {
-		Serial.println(configFile.fileSize());
-		byte buff[SERIAL_TX_BUFFER_SIZE] = {0};
-		int read;
-		do {
-			read = configFile.read(buff, SERIAL_TX_BUFFER_SIZE);
-			Serial.write(buff, read);
-		} while (read >= SERIAL_TX_BUFFER_SIZE);
-	}
-}
-
-void _renameTempFileToConfigFile(char const *path) {
-	if (SD.exists(path)) {
-		SD.remove(path);
-	}
-	configFile.rename(SD.vwd(), path);
-}
-
-void _openTempFile() {
-	if (SD.exists(TEMP_FILE)) {
-		SD.remove(TEMP_FILE);
-	}
-	configFile = SD.open(TEMP_FILE, O_WRONLY | O_CREAT);
-	configFile.seekSet(0);
-}
-
-long _getSerialFileSize() {
-	char numberChars[10];
-	FILL_BUFFER();
-	size_t len = Serial.readBytesUntil('\n', numberChars, 10);
-	numberChars[len] = '\n';
-	return atol(numberChars);
-}
-
-void saveNewConfigFileFromSerial() {
-	_openTempFile();
-	long fileSize = _getSerialFileSize();
-
-	long receivedBytes = 0;
-	unsigned int chunkLength;
-	FILL_BUFFER();
-	do {
-		byte input[SERIAL_RX_BUFFER_SIZE];
-		chunkLength = Serial.readBytes(input, SERIAL_RX_BUFFER_SIZE);
-		if (chunkLength == 0) break;
-		receivedBytes += chunkLength;
-		if (!(receivedBytes % 4096) || receivedBytes == fileSize)
-			Serial.println(receivedBytes);
-		configFile.write(input, chunkLength);
-	} while (chunkLength == SERIAL_RX_BUFFER_SIZE && receivedBytes < fileSize);
-	if (receivedBytes == fileSize) {
-		_renameTempFileToConfigFile(CONFIG_NAME);
-	}
-	configFile.close();
+  while (!SD.begin(SD_CS_PIN, SD_SCK_MHZ(16))) {
+    delay(1);
```

Comment:
This will create an infinite loop if there is something wrong with the SD card. Maybe print a message over the Serial that the SD card is missing. This way we have information if someone comes with issues that it won't boot